src/classes/BDI_DataImportService.cls

```diff
@@ -631,6 +639,29 @@ global with sharing class BDI_DataImportService {
         return fieldAccessLevelsToValidate;
     }
 
+    /**
+     * @description Before processing data import records, this method filters out any data imports that do not meet
+     * custom criteria that applies to all data imports before allowing them to be processed.
+     * @param dataImports {List<DataImport__c>} List of data import records to be filtered
+     *
+     * @return List<DataImport__c>
+    */
+    private List<DataImport__c> shouldProcess(List<DataImport__c> dataImports) {
+        List<DataImport__c> dataImportsToProcess = new List<DataImport__c>();
+
+        for (DataImport__c dataImport : dataImports) {
+            String paymentStatus = dataImport.Payment_Status__c;
+
+            if(!String.isNotBlank(paymentStatus) &&
+                dataImport.Payment_Status__c.toLowerCase() != PAYMENT_SERVICES_SUCCESSFUL_CHARGE) {
```

Comment:
@dfuller114 Will every DI record now be required to have a payment status of 'captured' to be processed ?

Reply:
good call. I'll update the if check there. forgot to remove the `!`. Should be checking for not blank and status 'CAPTURED` for the payment status. Meaning if the payment status is blank it will still process.

Reply:
Good call, Victor - I think this should be "if(String.isBlank(paymentStatus) || dataImport.Payment_Status__c.equalsIgnoreCase(PAYMENT_SERVICES_SUCCESSFUL_CHARGE)) {"

Reply:
@voduyemi fixed!

Reply:
@ptewson-sfdo i just updated it to `String.isNotBlank(paymentStatus) && dataImport.Payment_Status__c.toLowerCase() == PAYMENT_SERVICES_SUCCESSFUL_CHARGE`.

Reply:
@dfuller114 Haha, your brain's working :) but I don't think both statements are not the same, we expect that DI records with blank payment statuses to get processed, the statement you wrote will filter out those records.

Reply:
I agree that the "&&" really needs to be an "||" here. All records with a blank payment status field need to be processed. Unless I'm missing something, your logic as its written will only process them if the value in the payment status field is "captured".

I also prefer the use of "equalsIgnoreCase()" here over the "toLowerCase() ==" as it's more readable, and handles case incongruity in both arguments (I'm likely to also suggest updating the casing on the defined constant)

Reply:
Think of it this way - these values need to be processed:
 Payment Status: "" 
 Payment Status: "CAPTURED"

And this record should NOT be processed:
 Payment Status: "PENDING" 

For the blank status, String.isNotBlank(paymentStatus) returns "false" and `dataImport.Payment_Status__c.toLowerCase() == PAYMENT_SERVICES_SUCCESSFUL_CHARGE` also returns "false". False && False returns false and so the record won't be added to the list that's returned from the method, and won't be processed as a result.

Reply:
@ptewson-sfdo agreed. think i confused myself because originally I was going to not add to the list if the record shouldn't be processed so I changed that part without changing the if logic.

Reply:
just updated!